zcash_primitives/src/transaction/builder.rs

```diff
@@ -306,14 +440,76 @@ impl<'a, P: consensus::Parameters, R: RngCore> Builder<'a, P, R> {
             .ok_or(Error::InvalidAmount)
     }
 
+    pub fn compute_balances<BalanceError>(
```

Comment:
- Needs documentation.
- Document this addition in the changelog.
- This generic name is confusing because we _also_ import a `BalanceError` type. AFAICT we don't need it:
  ```suggestion
      pub fn compute_balances<S: FeeStrategy>(
  ```